sentry/src/main/java/io/sentry/util/TracingUtils.java

```diff
@@ -0,0 +1,106 @@
+package io.sentry.util;
+
+import io.sentry.Baggage;
+import io.sentry.BaggageHeader;
+import io.sentry.IHub;
+import io.sentry.ISpan;
+import io.sentry.PropagationContext;
+import io.sentry.Scope;
+import io.sentry.SentryOptions;
+import io.sentry.SentryTraceHeader;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class TracingUtils {
+
+  public static void startNewTrace(final @NotNull IHub hub) {
+    hub.configureScope(
+        scope -> {
+          scope.setPropagationContext(new PropagationContext());
+        });
+  }
+
+  public static void traceIfAllowed(
```

Comment:
M: I believe it'd be better if this just returns a `boolean`?
Executing a callback forces you to have Atomic operations in a few different places.

Reply:
Hmm could change it to return the `TracingHeaders` instead of having a callback. With `@Nullable` it'd only be a simple `null` check on the calling side. If this returned a boolean we'd have to duplicate the extraction code everywhere.

Reply:
Then returning `TracingHeaders` looks already better.